ext/pdo_pgsql/tests/gh12423.phpt

```diff
@@ -15,53 +20,55 @@ $dsnWithCredentials = $config['ENV']['PDOTEST_DSN'];
 $user = $config['ENV']['PDOTEST_USER'] ?? null;
 $password = $config['ENV']['PDOTEST_PASS'] ?? null;
 if (!$user) {
-    preg_match('/user=(.*?) /', $dsnWithCredentials, $match);
+    preg_match('/user=([^ ]*)/', $dsnWithCredentials, $match);
     $user = $match[1] ?? '';
 }
 if (!$password) {
-    preg_match('/password=(.*?)$/', $dsnWithCredentials, $match);
+    preg_match('/password=([^ ]*)/', $dsnWithCredentials, $match);
     $password = $match[1] ?? '';
 }
-$dsn = str_replace(" user={$user} password={$password}", '', $dsnWithCredentials);
+$dsn = str_replace("user={$user}", '', $dsnWithCredentials);
+$dsn = str_replace("password={$password}", '', $dsn);
+$dsn = rtrim($dsn);
 
 echo "dsn without credentials / correct user / correct password\n";
 try {
     $db = new PDO($dsn, $user, $password, [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]);
     echo "Connected.\n\n";
 } catch (PDOException $e) {
-    echo $e->getMessage();
+    echo $e->getMessage()."\n";
```

Comment:
nit: While at it, would be best to use PHP_EOL instead.

Reply:
Thank you, I fixed it